### DISCUSSION TITLE
Add source code in one commit seems impossible, but some have them

### DISCUSSION THREAD

**Discussion by ghost** [community/community#31162] [Other Feature Feedback, Questions, & Ideas] [answered]
A lot of initial commits add the source code. However it seems impossible (at least on web) because creating a repository does not have a section for source code. See [this commit for reference.](https://github.com/microsoft/playwright/commit/9ba375c06344835d783fe60bf33f857f9bc208a4). How does initial commits do this?

**Comment by airtower-luna** [comment, marked as answer]
You have the right idea already (emphasis added):

> However it seems impossible (**at least on web**)

Create your repository locally (`git init`), add and commit your files (`git add`, `git commit`), and push. Mind that you need to create the repository on GitHub empty (without an initial README or anything), because otherwise you'll run into conflicts.

If you don't feel familiar enough with Git on the command line yet, take a look at the ["Git Book" site](https://git-scm.com/book/en/v2), which explains it all in detail. Chapters 1 to 3 cover the basics. :slightly_smiling_face:
